test/python/tools/monitor/test_backend_monitor.py

```diff
@@ -14,43 +14,76 @@
 
 """Tests for the wrapper functionality."""
 
+import sys
 import unittest
 from unittest.mock import patch
+from unittest.mock import MagicMock
 from io import StringIO
 
+import qiskit
+from qiskit import providers
 from qiskit.tools.monitor import backend_overview, backend_monitor
-from qiskit.test import QiskitTestCase, online_test
+from qiskit.test import QiskitTestCase
+from qiskit.test.mock import FakeProvider
+from qiskit.test.mock import FakeBackend
+from qiskit.test.mock import FakeVigo
+
+
+FAKE_PROV = FakeProvider()
 
 
 class TestBackendOverview(QiskitTestCase):
     """Tools test case."""
 
-    @online_test
-    def test_backend_overview(self, qe_token, qe_url):
-        """Test backend_overview"""
-        from qiskit import IBMQ  # pylint: disable: import-error
-        IBMQ.enable_account(qe_token, qe_url)
-        self.addCleanup(IBMQ.disable_account)
+    def _restore_ibmq(self):
+        if not self.import_error:
+            qiskit.IBMQ = self.ibmq_back
+        else:
+            del qiskit.IBMQ
+        if self.prov_backup:
+            providers.ibmq = self.prov_backup
+        else:
+            del providers.ibmq
 
+    def setUp(self):
+        super().setUp()
+        ibmq_mock = MagicMock()
+        ibmq_mock.IBMQBackend = FakeBackend
+        sys.modules['qiskit.providers.ibmq'] = ibmq_mock
```

Comment:
Should this be restored during cleanup?

Reply:
yeah it probably should, I guess we were safe here because nothing is using the provider anymore so it doesn't matter. But yeah we should add a cleanup on this just in case

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/4296/commits/8718e6e450977fda04fad3686ea8baf326e2953d